services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java

```diff
@@ -111,4 +120,117 @@ public void checkIfSyncAttachmentState_isAvailable() {
         assertEquals(SyncAttachmentState.REDUCED_SYNC, absSyncViewModel.getCurrentSyncAttachmentState());
         assertNotNull(absSyncViewModel.getSyncAttachmentState().getValue());
     }
+
+    @Test
+    public void testNullAppName() {
+        absSyncViewModel.setAppName(null);
+        assertNull(absSyncViewModel.getAppName());
+    }
+
+    @Test
+    public void testNullServerUrl() {
+        absSyncViewModel.setServerUrl(null);
+        assertNull(absSyncViewModel.getServerUrl().getValue());
+    }
+
+    @Test
+    public void testNullUsername() {
+        absSyncViewModel.setUsername(null);
+        assertNull(absSyncViewModel.getUsername().getValue());
+    }
+
+    @Test
+    public void testIsFirstLaunchFlagTrue() {
```

Comment:
`testIsFirstLaunchFlagTrue`, `testIsFirstLaunchFlagFalse`: These tests essentially check the boolean getter for both possible values, I think a single test covering both cases would suffice.

Reply:
I'll combine these into one. Thanks for the feedback.